src/charm.py

```diff
@@ -69,14 +70,19 @@ def __init__(self, *args):
 
         self._storage_path = self.meta.storages["database"].location
 
-    def _redis_pebble_ready(self, _) -> None:
+    def _redis_pebble_ready(self, event) -> None:
         """Handle the pebble_ready event.
 
         Updates the Pebble layer if needed.
         """
         self._store_certificates()
         self._update_layer()
 
+        # update_layer will set a Waiting status if Pebble is not ready
+        if self.unit.status != ActiveStatus():
```

Comment:
I think this will evaluate to `False` if ActiveStatus has a status message - consider the following instead:
```suggestion
if isinstance(self.unit.status, ActiveStatus):
```

Reply:
You may also want to log why this event is being deferred

Reply:
Yes! You are right